transport_generic.go

```diff
@@ -26,10 +42,11 @@ func (t genericTransport) ReadMessage() (*Message, error) {
 }
 
 func (t genericTransport) SendMessage(msg *Message) error {
+	nativeEndian = NativeEndian()
```

Comment:
I wonder if the following code works for big endian machines. I don't have a big endian machine, can you check for me?
```
func detectEndianness() binary.ByteOrder {
	var x uint32 = 0x01020304
	if byte(x) == 0x01 {
		return binary.BigEndian
	}
	return binary.LittleEndian
}
```

If this works we can avoid using unsafe here.